rfcs/006-package-management.md

```diff
@@ -0,0 +1,243 @@
+---
+feature: package management
+start-date: 2024-06-28
+author: Joe Neeman
+---
+
+# Package management
+
+People want to reuse code, but Nickel doesn't currently have a good way to do
+it. We should have a way to fetch packages and make them easily available to
+Nickel code. The mechanism needs to be predictable (it should fetch the code
+that the user expects to fetch) and reliable (if it works on my machine it
+should work on your machine).
```

Comment:
It might sound obvious, but I think we might emphasize here clearly one of the (non)-goals: we only want to make Nickel code available to other Nickel code. We are not concerned with any kind of deeper system dependencies here, or cross-language ones (just use Nix or Bazel, I guess?)